### PR TITLE
Run kubemark presubmit on all PRs

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/29897

@gmarek - FYI

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/564)

### CODE REVIEW

File: ciongke/cmd/hook/main.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ var defaultJenkinsJobs = map[string][]JenkinsJob{
 		{
 			Name:      "kubernetes-pull-build-test-kubemark-e2e-gce",
 			Trigger:   regexp.MustCompile(`@k8s-bot kubemark (e2e )?test this`),
-			AlwaysRun: false,
+			AlwaysRun: true,
```
Comment (wojtek-t) [inline ciongke/cmd/hook/main.go]: I'm assuming that for now, those tests still won't be blocking right?
[Ideally I would like them to be blocking at some point, but we need to verify that it really works first.]

File: ciongke/cmd/hook/main.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ var defaultJenkinsJobs = map[string][]JenkinsJob{
...
 			Context:   "Jenkins Kubemark GCE e2e",
 		},
 		{
```
Comment (wojtek-t) [inline ciongke/cmd/hook/main.go]: What's the procedure to bump the image now?

**Comment by spxtr** [comment]
You need to bump the hook image version in the makefile as well as `hook_deployment.yaml`.

File: ciongke/cmd/hook/main.go (hunk 1 above)
Comment (spxtr) [inline ciongke/cmd/hook/main.go]: Correct. What is and isn't blocking is actually determined by the munger.

**Comment by spxtr** [comment]
Will need to rope in @rmmh since I don't think we have enough Jenkins machines for this right now.

**Comment by spxtr** [comment]
It also looks like that test has been pretty red recently, which is worrying.

**Comment by rmmh** [comment]
Is this the full kubemark test? I don't think we have enough cores to run that on all presuming. We could probably handle a smoke test. 

I'm also against making this a blocking job immediately. We should run it for a while and shake out any flakes first.

**Comment by gmarek** [comment]
Depends on the definition of "real" - it's equivalent to kubemark-5 and requires 4 cores total (1 core master, 2 core node, and 1 core kubemark-master)

**Comment by spxtr** [comment]
You'll need enough quota in the project for 12 concurrent runs, and we'll need to add 12 more Jenkins agents.

**Comment by rmmh** [comment]
Do we really need to run as many agents as we could possibly have jobs
running? Queuing works fine.

**Comment by spxtr** [comment]
Sure, but we'll get backed up a little at certain times. It's not a big deal either way, I don't think. Maybe 8-10 agents per per-pr test is fine.

**Comment by wojtek-t** [comment]
>
It also looks like that test has been pretty red recently, which is worrying.

@spxtr - yes, because someone broke it - this is what we would like to avoid.

>
Is this the full kubemark test? I don't think we have enough cores to run that on all presubmits. We could probably handle a smoke test.

It's relatively small test - from Jenkins pov it's super light.

>
I'm also against making this a blocking job immediately. We should run it for a while and shake out any flakes first.

Totally agree - I would like to make it blocking once we are sure it works correctly.

>
You'll need enough quota in the project for 12 concurrent runs,

we already have it

>
 and we'll need to add 12 more Jenkins agents.

What is Jenkins agent. The test is pretty light itself.

**Comment by spxtr** [comment]
> What is Jenkins agent? The test is pretty light itself.

We just need to turn up a few more Jenkins VMs since we only run one PR job per VM. We used to get lots of flakes before we made that change, but it means we need to turn up VMs when we add jobs.

**Pull request merged**